libantworld/agent.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+// OpenGL includes
+#include <GL/glew.h>
+
+// GLFW
+#include <GLFW/glfw3.h>
+
+// Libantworld includes
+#include "common.h"
+#include "renderer.h"
+
+// BoB robotics includes
+#include "../common/pose.h"
+#include "../video/opengl.h"
+
+using namespace BoBRobotics::Pose;
+
+namespace BoBRobotics {
+namespace AntWorld {
+template<class LengthUnit = meter_t, class AngleUnit = radian_t>
+class AntAgent
+  : Positionable<LengthUnit>
+  , Rotatable<AngleUnit>
+  , Video::OpenGL
+{
+public:
+    AntAgent(GLFWwindow *window, Renderer &renderer, GLsizei readWidth, GLsizei readHeight);
+    Vector<LengthUnit> &getPosition() override;
+    Vector<AngleUnit> &getAttitude() override;
+    void setPosition(LengthUnit x, LengthUnit y, LengthUnit z) override;
+    void setAttitude(AngleUnit yaw, AngleUnit pitch, AngleUnit roll) override;
+    bool readFrame(cv::Mat &frame) override;
+
+private:
+    Vector<AngleUnit> m_Attitude{ 0_rad, 0_rad, 0_rad };
+    Vector<LengthUnit> m_Position{ 0_m, 0_m, 0_m };
+    Renderer &m_Renderer;
+    GLFWwindow *m_Window;
+}; // AntAgent
+} // AntWorld
+} // BoBRobotics
+
+namespace BoBRobotics {
```

Comment:
Especially with templates, I find this inlining syntax really cumbersome and verbose. Is there any reason not to have the methods inline as in where the class is declared? I always felt that was one of the side-benefits of header-only :)

Reply:
Yeah you're right :+1: